File: library/src/main/kotlin/com/ohmerhe/kolley/image/ImageDisplayOption.java
package com.ohmerhe.kolley.image;

import android.graphics.Bitmap;
import android.widget.ImageView;

/**
 * Created by ohmer on 1/12/16.
 */
public class ImageDisplayOption {
    public final static int DETAULT_IMAGE_WIDTH_MAX = 1080;
    public final static int DETAULT_IMAGE_HEIGHT_MAX = 1290;
    private final int imageResOnLoading;
    private final int imageResForEmptyUri;
    private final int imageResOnFail;
    private final Bitmap.Config decodeConfig;
    private final ImageView.ScaleType scaleType;
    private final int maxWidth;
    private final int maxHeight;

    private ImageDisplayOption(Builder builder) {
        imageResOnLoading = builder.imageResOnLoading;
        imageResForEmptyUri = builder.imageResForEmptyUri;
        imageResOnFail = builder.imageResOnFail;
        decodeConfig = builder.decodeConfig;
        scaleType = builder.scaleType;
        maxWidth = builder.maxWidth;
        maxHeight = builder.maxHeight;
    }


    public static final class Builder {
        private int imageResOnLoading;
        private int imageResForEmptyUri;
        private int imageResOnFail;
        private Bitmap.Config decodeConfig = Bitmap.Config.RGB_565;
        private ImageView.ScaleType scaleType = ImageView.ScaleType.CENTER_CROP;
        private int maxWidth = DETAULT_IMAGE_WIDTH_MAX;
        private int maxHeight = DETAULT_IMAGE_HEIGHT_MAX;

        public Builder() {
        }

        public Builder imageResOnLoading(int val) {
            imageResOnLoading = val;
            return this;
        }

        public Builder imageResForEmptyUri(int val) {
            imageResForEmptyUri = val;
            return this;
        }

        public Builder imageResOnFail(int val) {
            imageResOnFail = val;
            return this;
        }

        public Builder decodeConfig(Bitmap.Config val) {
            decodeConfig = val;
            return this;
        }

        public Builder scaleType(ImageView.ScaleType val) {
            scaleType = val;
            return this;
        }

        public Builder maxWidth(int val) {
            maxWidth = val;
            return this;
        }

        public Builder maxHeight(int val) {
            maxHeight = val;
            return this;
        }

        public ImageDisplayOption build() {
            return new ImageDisplayOption(this);
        }
    }

    public int getImageResOnLoading() {
        return imageResOnLoading;
    }

    public int getImageResForEmptyUri() {
        return imageResForEmptyUri;
    }

    public int getImageResOnFail() {
        return imageResOnFail;
    }

    public Bitmap.Config getDecodeConfig() {
        return decodeConfig;
    }

    public ImageView.ScaleType getScaleType() {
        return scaleType;
    }

    public int getMaxWidth() {
        return maxWidth;
    }

    public int getMaxHeight() {
        return maxHeight;
    }
}

File: library/src/main/kotlin/com/ohmerhe/kolley/image/ImageDisplayOption.kt
/*
 * Copyright (c) 2016  Ohmer.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.ohmerhe.kolley.image

import android.graphics.Bitmap
import android.widget.ImageView

/**
 * Created by ohmer on 1/12/16.
 */
class ImageDisplayOption private constructor(builder: ImageDisplayOption.Builder) {
    val imageResOnLoading: Int
    val imageResForEmptyUri: Int
    val imageResOnFail: Int
    val decodeConfig: Bitmap.Config
    val scaleType: ImageView.ScaleType
    val maxWidth: Int
    val maxHeight: Int

    init {
        imageResOnLoading = builder.imageResOnLoading
        imageResForEmptyUri = builder.imageResForEmptyUri
        imageResOnFail = builder.imageResOnFail
        decodeConfig = builder.decodeConfig
        scaleType = builder.scaleType
        maxWidth = builder.maxWidth
        maxHeight = builder.maxHeight
    }


    class Builder {
        var imageResOnLoading: Int = 0
        var imageResForEmptyUri: Int = 0
        var imageResOnFail: Int = 0
        var decodeConfig = Bitmap.Config.RGB_565
        var scaleType = ImageView.ScaleType.CENTER_CROP
        var maxWidth = DETAULT_IMAGE_WIDTH_MAX
        var maxHeight = DETAULT_IMAGE_HEIGHT_MAX

        fun build(): ImageDisplayOption {
            return ImageDisplayOption(this)
        }
    }

    companion object {
        val DETAULT_IMAGE_WIDTH_MAX = 1080
        val DETAULT_IMAGE_HEIGHT_MAX = 1960
    }
}
